### DISCUSSION TITLE
Version 2.0 Incoming! 🎉

### DISCUSSION THREAD

**Discussion by E-Kuerschner** [E-Kuerschner/useAudioPlayer#106] [Announcements]
### Update => v2.0.0-alpha.0 ready to test! (https://github.com/E-Kuerschner/useAudioPlayer/discussions/108)

👋 Greetings useAudioPlayer users!!

I just wanted to get notice out that I'm working on a full rewrite of the package which I expect to drop some time in May. Below are a few things of note about the new version. Please leave any comments/concern/feedback in the thread. Cheers!

### useAudioPlayer -> useGlobalAudioPlayer
The original use case I set out to satisfy with this package was a singular audio source which could be controlled from anywhere in your app/component tree. This was inspired by applications like Spotify or SoundCloud. Because of the frequent requests to support multiple audio sources from the community, I will be renaming the main hook to better convey what it does, which is controlling a global audio source for the whole app.

### no more AudioPlayerContext
One of the major complaints I had with the original implementation was the need to wrap the portion of the app which controls audio in a Provider. This was a constant source of bugs and did not feel intuitive to write. You will be happy to hear that `useGlobalAudioPlayer` will be supported _without_ the need for a Provider wrapping its usages 🎉 

### multiple audio sources
The name of the original hook, `useAudioPlayer` will now be used for a new hook. A component can instantiate multiple instances of this hook to simultaneously control different audio sources.

### no more useAudioPosition
This one may come as a surprise to some, and I apologize, but it does feel like the responsibility of this package to provide a solution to animation (animating a seek bar/time code), this can easily be achieved in the client application without too much extra effort. I still believe that a constant state update for every change in audio progress is to heavy to include in the main hook (this is why useAudioPosition was created originally), so there will only be a getter function provided on the API, `getPosition`.

**Comment by no-1ne** [comment]
looking forward espl being able to play multiple audio sources, 

i have a feature request 

- being able  to control individual audio's volume, 
- being able to add a new audio source which returns a audioRefid, 
- we can use this to refID, to control the volume & turn off the audio

**Reply by E-Kuerschner** [reply]
i think the working draft for v2 should support all your feature requests. Ill notify here when i publish an alpha version so you can test it out and leave feedback

**Reply by E-Kuerschner** [reply]
@no-1ne v2 alpha is ready for testing
https://github.com/E-Kuerschner/useAudioPlayer/discussions/108

**Reply by geniux-dev** [reply]
When the audio is playing, the browser takes a while to open a page. i think it might be using too much memory, am using react nextjs

**Reply by E-Kuerschner** [reply]
@geniux-dev can you pull together a reliable way to reproduce the experience? Or put together a sandbox where the issue is prevent? I have not run into similar issues on my end yet

**Comment by aweibell** [comment]
Great to see that you pick upt this project and take it further!

Interesting to read about your ideas. I'm curious to see how it will fit my use case.

Not sure if you intend to support multiple simultaneously playing audio sources, or multiple to switch between. I guess  both concepts can be relevant for different apps. Multiple tracks playing at the same time would certainly confuse my users :)

I definitely need to keep track(!) of the position, so I will have to find a way for that. Maybe it can be done with a simple `setInterval` to call `getPosition` every second?

**Reply by no-1ne** [reply]
html media element exposes currentTime, may be query it on demand vs polling via setInterval  https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/currentTime

**Reply by E-Kuerschner** [reply]
@aweibell you are right. Without the useAudioPosition library you can set your own interval and poll the `getPosition` method which the hooks in v2 will expose.

In v2 you will be able to support either a single, global audio source with `useGlobalAudioPlayer` or multiple sound instances with `useAudioPlayer`, whichever better suites your use case!

**Reply by E-Kuerschner** [reply]
@aweibell v2 alpha is ready for testing!
https://github.com/E-Kuerschner/useAudioPlayer/discussions/108

**Reply by aweibell** [reply]
> @aweibell v2 alpha is ready for testing! #108

@E-Kuerschner Thank you! I saw your announcement :) Just have not had the time to look at it yet :-/

**Reply by phemartin** [reply]
This is awesome @E-Kuerschner - I can't wait to test it out!
